### PR TITLE
Setup associations

### PR DESCRIPTION
**Pull request by kensteph**
Hi there 👋!
In this PR I : 
✅ Create a Classroom class and its constructor.
✅ Create Setter and Getter for @label.
✅ Create the has-many relationship between Classroom and Student.
✅ Create the belongs-to relationship between Classroom and Student.
✅ Make sure that when adding a student to a classroom it also sets the classroom for the student. And Make sure that when setting the classroom for a student it also adds it to the classroom's students.
✅ Create a Book class and its constructor.
✅ Create a Getter and Setter for the title and author.
✅ Create a Rental class and its constructor.
✅ Create a Getter and Setter for a date.
✅ Create the has-many side of Book and Rental.
✅ Create the belongs-to side of Rental and Book.
✅ Create the has-many side of Person and Rental.
✅ Create the belongs-to side of Rental and Person.
✅ Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,17 @@
 require_relative './person'
 
 class Student < Person
+  attr_reader :classroom
+
   def initialize(age, classroom, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    @classroom.students.push(self) unless @classroom.students.include?(self)
```
Comment (mckent05) [inline student.rb]: - This line of code is not necessary, you did implementing this in your customer setter. Please remove this 🙏

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+  end
+end
```
Comment (mckent05) [inline rental.rb]: - You are yet to implement the belongs-to relationship between the `Rental` class and the `Book` and `Person` class. Remember that a `rental` belongs to both `Book` and `person` i.e `book.rentals << self`, `person.rentals <<self`. Please fix this 🙏

File: person.rb [hunk 3]
```diff
@@ -29,4 +30,8 @@ def can_use_services?
   def correct_name
     @name
   end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```
Comment (mckent05) [inline person.rb]: - You have not properly implemented the has-many relationship between the `Person` class and the `Rental` class. Remember a `person` can have many `rentals` i.e the `add_rental` method should look like this
``` 
def add_rental( date, book)
   Rental.new(date, book, self)
end
```

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rentals.push(rental)
+  end
```
Comment (mckent05) [inline book.rb]: - You have not properly implemented the has-many relationship between the `Book` class and the `Rental` class. Remember a `book` can have many rentals i.e the add_rental method should look like this
```
def add_rental( date, person)
   Rental.new(date, self, person)
end
```

Comment (mckent05) [inline book.rb]: - You have not properly implemented the has-many relationship between the `Book` class and the `Rental` class. Remember a `book` can have many rentals i.e the add_rental method should look like this
```
def add_rental( date, person)
   Rental.new(date, self, person)
end
```

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @kensteph, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job setting up associations between your classes


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kensteph** [comment]
Thank you @mckent05 for your review.

**Review by danmainah — APPROVED** [review]
Hi @kensteph , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**